### PR TITLE
TPC QC: add task for SACs

### PR DESCRIPTION
**Pull request by tklemenz**
This adds a QC task for the SACs.

### CODE REVIEW

File: Detectors/TPC/qc/include/TPCQC/SACs.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  TCanvas* drawSACTypeSides(const SACType type, const unsigned int integrationInterval, const int minZ = 0, const int maxZ = -1, TCanvas* canv = nullptr);
+  TCanvas* drawSACOneCanvas(int nbins1D, float xMin1D, float xMax1D, int integrationIntervals = -1, TCanvas* outputCanvas = nullptr) const;
+  TCanvas* drawFourierCoeffSAC(Side side, int nbins1D, float xMin1D, float xMax1, TCanvas* outputCanvas = nullptr) const;
+
+  void dumpToFile(std::string filename, int type = 0);
+
+ private:
+  std::array<std::vector<int32_t>, o2::tpc::GEMSTACKS> mSACs{};
+  SACZero mSACZero{};
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/SACs.h]: Do we need a copy?

File: Detectors/TPC/qc/include/TPCQC/SACs.h [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  std::array<SACOne*, SIDES> mSACOne = {nullptr, nullptr}; ///< I_1(t) = <I(r,\phi,t) / I_0(r,\phi)>_{r,\phi}
+  SACDelta<unsigned char> mSACDelta{};
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/SACs.h]: Same here. Do we need a copy?

File: Detectors/TPC/qc/src/SACs.cxx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+#include "fmt/format.h"
+
+ClassImp(o2::tpc::qc::SACs);
+using namespace o2::tpc::qc;
+
+float SACs::getSACOneVal(const Side side, unsigned int integrationInterval) const
+{
+  return !mSACOne[side] ? -1 : mSACOne[side]->getValue(side, integrationInterval);
+}
+TCanvas* SACs::drawSACTypeSides(const SACType type, const unsigned int integrationInterval, const int minZ, const int maxZ, TCanvas* canv)
```
Comment (wiechula) [inline Detectors/TPC/qc/src/SACs.cxx:25]: empty line missing
```suggestion

TCanvas* SACs::drawSACTypeSides(const SACType type, const unsigned int integrationInterval, const int minZ, const int maxZ, TCanvas* canv)
```

File: Detectors/TPC/qc/src/SACs.cxx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  SACDrawHelper::SACDraw drawFun;
+  drawFun.mSACFunc = SACFunc;
+  const std::string zAxisTitle = SACDrawHelper::getZAxisTitle(type);
+
+  auto hSideA = SACDrawHelper::drawSide(drawFun, o2::tpc::Side::A, zAxisTitle);
+  auto hSideC = SACDrawHelper::drawSide(drawFun, o2::tpc::Side::C, zAxisTitle);
+
+  hSideA->SetTitle(fmt::format("{} ({}-Side)", name.data(), "A").data());
+  hSideC->SetTitle(fmt::format("{} ({}-Side)", name.data(), "C").data());
+
```
Comment (wiechula) [inline Detectors/TPC/qc/src/SACs.cxx:60]: Do we need `hSideA/C->SetBit(TObject::kCanDelete);`?

File: Detectors/TPC/qc/src/SACs.cxx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+{
+  auto* canv = outputCanvas;
+
+  if (!canv) {
+    canv = new TCanvas("c_sides_SAC1_1D", "SAC1 1D distribution for each side", 1000, 1000);
+  }
+
+  auto hAside1D = new TH1F("h_SAC1_1D_ASide", "SAC1 distribution over integration intervals A-Side", nbins1D, xMin1D, xMax1D);
+  auto hCside1D = new TH1F("h_SAC1_1D_CSide", "SAC1 distribution over integration intervals C-Side", nbins1D, xMin1D, xMax1D);
+
```
Comment (wiechula) [inline Detectors/TPC/qc/src/SACs.cxx:90]: `hA/Cside1D->SetBit(TObject::kCanDelete);` ?

File: Detectors/TPC/qc/src/SACs.cxx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  canv->DivideSquare(mFourierSAC->mCoeff[side].getNCoefficientsPerTF());
+
+  size_t pad = 1;
+
+  for (const auto& hist : histos) {
+    canv->cd(pad);
+    hist->SetTitleOffset(1.05, "XY");
+    hist->SetTitleSize(0.05, "XY");
+    hist->Draw();
```
Comment (wiechula) [inline Detectors/TPC/qc/src/SACs.cxx:146]: `->SetBit(TObject::kCanDelete);`?

File: Detectors/TPC/qc/include/TPCQC/SACs.h (hunk 1 above)
Comment (tklemenz) [reply Detectors/TPC/qc/include/TPCQC/SACs.h]: No, I changed it. Same below.

**Comment by tklemenz** [comment]
@wiechula Can this be merged?

**Review by wiechula — APPROVED** [review]

**Comment by wiechula** [comment]
Yes, fine from my side.

**Pull request merged**